src/test/java/org/folio/rest/api/HridSettingsIncreaseMaxValueMigrationTest.java

```diff
@@ -0,0 +1,120 @@
+package org.folio.rest.api;
+
+import static org.folio.rest.api.StorageTestSuite.TENANT_ID;
+import static org.folio.rest.api.StorageTestSuite.getVertx;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.folio.rest.persist.PostgresClient;
+import org.folio.util.ResourceUtil;
+import org.junit.Test;
+
+import io.vertx.core.json.JsonArray;
+
+public class HridSettingsIncreaseMaxValueMigrationTest extends TestBase {
+  private static final String CREATE_SEQUENCE_SQL = loadCreateSequenceSqlFile();
+  private static final String ALTER_SEQUENCE_SQL = loadAlterSequenceSqlFile();
+  private static final String INSTANCES_SEQ = "hrid_instances_seq";
+  private static final String HOLDINGS_SEQ = "hrid_holdings_seq";
+  private static final String ITEMS_SEQ = "hrid_items_seq";
+
+  @Test
+  public void retainsCurrentValueOfSequences() throws Exception {
+    reCreateSequences();
+
+    assertThat(incrementSequence(INSTANCES_SEQ, 10), is(10L));
+    assertThat(incrementSequence(HOLDINGS_SEQ, 7), is(7L));
+
+    setValue(ITEMS_SEQ, 999_999_99L);
+    assertThat(incrementSequence(ITEMS_SEQ), is(999_999_99L));
+
+    alterSequences();
+
+    assertThat(incrementSequence(INSTANCES_SEQ), is(11L));
+    assertThat(incrementSequence(HOLDINGS_SEQ), is(8L));
+    assertThat(incrementSequence(ITEMS_SEQ), is(1_000_000_00L));
+  }
+
+  private void reCreateSequences() throws Exception {
+    executeSql(dropSequenceSql(INSTANCES_SEQ));
+    executeSql(dropSequenceSql(HOLDINGS_SEQ));
+    executeSql(dropSequenceSql(ITEMS_SEQ));
+
+    executeSqlFile(CREATE_SEQUENCE_SQL);
+  }
+
+  private void alterSequences() throws InterruptedException, ExecutionException,
+    TimeoutException {
+
+    executeSqlFile(ALTER_SEQUENCE_SQL);
+  }
+
+  private Long incrementSequence(String sequenceName)
+    throws InterruptedException, ExecutionException, TimeoutException {
+
+    final CompletableFuture<Long> nextValue = new CompletableFuture<>();
+    final String sql = String.format(
+      "SELECT nextVal('%s.%s')", getSchemaName(), sequenceName
+    );
+
+    PostgresClient.getInstance(getVertx()).select(sql, result -> {
+      if (result.failed()) {
+        nextValue.completeExceptionally(result.cause());
+      } else {
+        List<JsonArray> results = result.result().getResults();
+        if (results.isEmpty()) {
+          nextValue.completeExceptionally(new RuntimeException("Empty result set"));
+        } else {
+          nextValue.complete(results.get(0).getLong(0));
+        }
+      }
+    });
+
+    return nextValue.get(5, TimeUnit.SECONDS);
+  }
+
+  private Long incrementSequence(String sequenceName, int times)
```

Comment:
Is this function used?

Reply:
yes, see HridSettingsIncreaseMaxValueMigrationTest lines # 31-32

Reply:
Ok